packages/applicant/src/middleware.page.ts

```diff
@@ -135,6 +148,40 @@ export async function middleware(req: NextRequest) {
   }
 }
 
+async function getApplicationStatusBySubmissionId(
+  submissionId: string,
+  jwt: string
+) {
+  const data = await fetch(
+    `${BACKEND_HOST}/submissions/${submissionId}/application/status`,
+    {
+      headers: {
+        Authorization: `Bearer ${jwt}`,
+        Accept: 'application/json',
+      },
+    }
+  );
+  return await data.text();
+}
+
+async function shouldRedirectToClosedGrantPage(jwt: string, req: NextRequest) {
+  const { pathname } = req.nextUrl;
+  const submissionId = pathname.split('/')[2];
+  console.log('submissionId', submissionId ? submissionId : 'no submissionId');
+  console.log('pathname', pathname);
+
+  if (submissionId) {
+    const applicationStatus = await getApplicationStatusBySubmissionId(
+      submissionId,
+      jwt
+    );
+    if (applicationStatus === 'REMOVED') {
```

Comment:
probably easier to read this if we early return falsy when there is no `submissionId` (instead of nesting ifs)

Reply:
this will probably cause issues for [GAP-2389: Applicant views answers of partially completed application after the grant closes](https://technologyprogramme.atlassian.net/browse/GAP-2389)